### ISSUE TITLE
[Bug] 'Image' field is inmutable but should be mutable

### ISSUE THREAD

**Issue by outductor** [mariadb-operator/mariadb-operator#143]
**Describe the bug**

```diff
spec:
  image:
-  tag: "10.11.3"
+  tag: "10.11.4"
```

```
admission webhook "vmariadb.kb.io" denied the request: MariaDB.mariadb.mmontes.io "mariadb" is invalid: Image: Invalid value: v1alpha1.Image{Repository:"mariadb", Tag:"10.11.4", PullPolicy:"IfNotPresent"}: 'Image' field is inmutable
```

**Expected behaviour**

MariaDB version should be upgradable

**Steps to reproduce the bug**

**Additional context**

**Environment details**:
- Kubernetes version:
- mariadb-operator version: 0.15.1
- Install method: helm, OLM or static manifests
- Install flavour: minimal, recommended or custom

**Comment by mmontes11** [comment]
Hey @outductor ! thanks for bringing this up.

The image being inmutable is intentional, as we don't currently support version upgrades in the operator:
- https://github.com/mariadb-operator/mariadb-operator/issues/118

However, we could make it mutable and support the [updateStrategy](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#update-strategies), something that might be valid for your case, as there should be no breaking changes when upgrading a patch version like in this case.e

**Comment by mmontes11** [comment]
Hey there! this has been supported via the following PR:
- https://github.com/mariadb-operator/mariadb-operator/pull/150

It will be released in `v0.0.16`! Stay tuned! 🚀
